### PR TITLE
AAP-51502 Make authenticator.configuration reqd

### PR DESCRIPTION
**Pull request by bhavenst**
* The model lists the field as blank=True but it is required by the validator (blank is OK for some auth types)
 * Required for all operations except PATCH
 * Fixes openapi spec/ATF
 * Pins drf-spectacular in dev to the same version used by PDE (aligns spec file output)

## Description
- What is being changed?  "configuration" field made required in serializer
- Why is this change needed? to fix openapi spec for ATF purposes
- How does this change address the issue? DRF spectacular respects the serializer configuration, showing configuration as required

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test update
- [ ] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change

## Self-Review Checklist
- [x] I have performed a self-review of my code
- [x] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment

### Steps to Test
1. Start gateway, load docs from /api/gateway/v1/docs/
2. See that configuration field is shown as required for all operations except PATCH
3. 

### Expected Results
Openapi spec is correct.

### CODE REVIEW

File: ansible_base/authentication/serializers/authenticator.py [hunk 1]
```diff
@@ -35,7 +35,10 @@ class Meta:
             "slug": {
                 "allow_blank": True,
                 "required": False,
-            }
+            },
+            "configuration": {
```
Comment (Dostonbek1) [inline ansible_base/authentication/serializers/authenticator.py:39]: Looking at line 110, it says `# Not having configuration is only valid for a PATCH`; seems like we are not requiring the field for PATCH method. Would this change break that logic?

Comment (bhavenst) [reply ansible_base/authentication/serializers/authenticator.py:39]: No, DRF spectacular knows that fields are not required for PATCH even if they are for other actions.

**Review by Dostonbek1 — APPROVED** [review]

**Review by john-westcott-iv — APPROVED** [review]
